crates/bevy_sprite/src/texture_atlas.rs

```diff
@@ -117,40 +107,60 @@ impl TextureAtlas {
 
         let grid_size = Vec2::new(columns as f32, rows as f32);
 
-        TextureAtlas {
+        Self {
             size: ((tile_size + current_padding) * grid_size) - current_padding,
             textures: sprites,
-            texture,
             texture_handles: None,
         }
     }
 
-    /// Add a sprite to the list of textures in the [`TextureAtlas`]
-    /// returns an index to the texture which can be used with [`TextureAtlasSprite`]
+    /// Add a *section* to the list in the layout and returns its index
+    /// which can be used with [`TextureAtlas`]
     ///
     /// # Arguments
     ///
-    /// * `rect` - The section of the atlas that contains the texture to be added,
-    /// from the top-left corner of the texture to the bottom-right corner
+    /// * `rect` - The section of the texture to be added
+    ///
+    /// [`TextureAtlas`]: crate::TextureAtlas
     pub fn add_texture(&mut self, rect: Rect) -> usize {
         self.textures.push(rect);
         self.textures.len() - 1
     }
 
-    /// The number of textures in the [`TextureAtlas`]
+    /// How many textures are in the `TextureAtlas`
```

Comment:
Might be clearer to say that this function "returns the amount...". I'm not familiar with any docs comment conventions though...